examples/bert/src/bert_layers.py

```diff
@@ -474,7 +483,18 @@ def forward(
         #        hidden_states[ntokens,hidden] -> hidden_states[ntokens_unpad,hidden]
         hidden_states, indices, cu_seqlens, _ = unpad_input(
             hidden_states, attention_mask_bool)
+
         # Add alibi matrix to extended_attention_mask
+        if self._current_alibi_size < seqlen:
+            # Build/rebuild the alibi tensor if needed
+            self.alibi = self.build_alibi_tensor(
+                n_heads=self.num_attention_heads,
+                size=seqlen,
+                device=hidden_states.device)
+            self._current_alibi_size = int(seqlen)
+        elif self.alibi.device != hidden_states.device:
+            # Device catch-up
+            self.alibi = self.alibi.to(hidden_states.device)
         alibi_bias = self.alibi[:, :, :seqlen, :seqlen]  # type: ignore
         alibi_attn_mask = extended_attention_mask[:, :, :, :seqlen] + alibi_bias
```

Comment:
I'm technically future proofing for if extended_attention_mask is ever extended to be have shape [b, h, seqlen, seqlen] since currently it has shape [b, 1, 1, seqlen] but even if shape is [b, 1, 1, seqlen], `[:, :, :seqlen, :seqlen]` should work because python / pytorch is smart that way (at least I've done it in the past to tensors which could have either shape).
Given that this is a "future proofing" feel free to ignore this suggestion.